### PR TITLE
Fix for #315. Sets zone based on approximateZoneFromeHostname

### PR DESCRIPTION
**Pull request by ryanjbaxter**
See #315

### CODE REVIEW

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/eureka/ZoneUtils.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+public class ZoneUtils {
+
+	/**
+	 * Approximates Eureka zones from a host name. This method approximates the zone to be
+	 * everything after the first "." in the host name.
+	 * @param host The host name to extract the host name from
+	 * @return The approximate zone
+	 */
+	public static String extractApproximateZone(String host) {
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/eureka/ZoneUtils.java]: How about an interface and implementation here?  I can imagine there might be other ways to approximate zone.  What do you think?

Comment (ryanjbaxter) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/eureka/ZoneUtils.java]: Other ways to approximate the zone from the host name?  Right now there is only a property to enable this functionality based off the host name.  If you think there might be more than one way of doing that than I think an interface might make sense.

Comment (dsyer) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/eureka/ZoneUtils.java]: I'm happy to merge this as a bug fix for now. Please send the interface idea as a separate pull request if you have time.
